### PR TITLE
Best/worst defense tt

### PR DESCRIPTION
**Pull request by tylertomlinson**
Final methods for iteration 3 - Best defense  /  Worst defense 
Created helper methods : 
home_id_defense stats - creates hash { home_id => away _goals }
away_id_defense stats - creates hash { away_id => home _goals }
defense_helper - takes above two hashes to create singe hash and merges identical 
    keys together while adding the values. 

Best defense and Worst defense take the defense helper method and will either max_by or min_by. The result is an array with two objects. First being the team_id and second the goals. Calling first on that array will return the team_id. Being able to call find_name on the team_id will pass spec_harness.

Will combine the first two helper methods and pass arguments within defense helper method later to reduce lines. 

Tests created for all methods
Rake passes with SimpleCov covering 96.1%

### CODE REVIEW

File: lib/league_stats.rb [hunk 1]
```diff
@@ -71,6 +71,16 @@ def worst_fans
     team_names
   end
 
+  def best_defense
+    best_defense_id = defense_helper.max_by { |id, goals| goals }.first
+    find_name(best_defense_id)
+  end
```
Comment (whitneykidd) [inline lib/league_stats.rb]: ~*so fresh and so clean, clean*~

File: lib/league_stats.rb [hunk 2]
```diff
@@ -188,4 +198,22 @@ def games_by_team(team_id)
   def total_games_by_team_id(team_id)
     games_by_team(team_id).length
   end
+
+
+  def home_id_defense_stats
```
Comment (whitneykidd) [inline lib/league_stats.rb:203]: ⚡️ All of these methods look so sleek ⚡️

File: lib/league_stats.rb [hunk 3]
```diff
@@ -188,4 +198,22 @@ def games_by_team(team_id)
...
+    @games.group_by(&:home_team_id)
+    .map{ |id, away_goals| [id, away_goals.map(&:away_goals).inject(:+)] }.to_h
+  end
+
+  def away_id_defense_stats
+    @games.group_by(&:away_team_id)
+    .map{ |id, away_goals| [id, away_goals.map(&:home_goals).inject(:+)] }.to_h
+  end
+
+  def defense_helper
```
Comment (Maxwell-Baird) [inline lib/league_stats.rb:213]: Your helper looks awesome

File: test/league_stats_test.rb [hunk 4]
```diff
@@ -138,4 +138,34 @@ def test_game_teams_stats_update_id
     expected = {'id' => [5.5, 2]}
     assert_equal expected, @league_stats.update_id(id, key, test_hash)
   end
+
+  def test_returns_home_id_and_away_goals
+    #First object within each array is home_id.
```
Comment (alex-latham) [inline test/league_stats_test.rb:143]: Array or hash?

**Pull request merged**